core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -357,7 +357,13 @@ private[spark] class BlockManager(
       info: BlockInfo,
       status: BlockStatus,
       droppedMemorySize: Long = 0L): Unit = {
-    val needReregister = !tryToReportBlockStatus(blockId, info, status, droppedMemorySize)
+    def ifAsync: Boolean = {
+      blockId.isBroadcast
+    }
+    // asynchronously report broadcast variables, in future we only need to change the
+    // implementation of ifAsync
```

Comment:
I find this comment a little bit confusing -- it sounds like you are saying there is future work still to do.  Do you mean something like "Broadcast variables are currently reported asynchronously, but if we later decide they should be reported synchronously, we just need to update ifAsync`"